### ISSUE TITLE
Make incoming call screen more functional, propose: answer, reject, reject with sms, mute

### ISSUE THREAD

**Issue by vaspa** [ossw/ossw-android#25]
Different Android versions and manufacturer customized ones have different implementations for call notifications. It may appear that implementing a separate feature based on phone state (TelephonyManager, PhoneStateListener, etc.) with proper notifications can be more consistent.

**Comment by althink** [comment]
I'm afraid that programatically answering the phone is not part of public API. There are many workarounds that work on certain android versions but none is guaranteed to work on every phone:

http://stackoverflow.com/questions/26924618/how-can-incoming-calls-be-answered-programmatically-in-android-5-0-lollipop

It looks like none of them may work on HTC phones:
https://forums.xamarin.com/discussion/36102/how-to-answer-a-phone-call-programmatically

In old version https://github.com/ossw/ossw-android/commit/06c009232295385304933c28458668dd80919bb0 it was a separate feature but I had problem with answering calls. If you find a good way that work on all phones then it will be great. I've decided to use notification because at the time I thought that's the most generic option (and it may appear that it's not)

**Comment by McBoby** [comment]
yeah the incoming Call Id can it Also have a Geolocation Of the Caller Displayed on, think it will be best, if anyone has an Idea plss can you hint me up. thanks
